contrib/src/main/java/com/datatorrent/contrib/avro/AvroToPojo.java

```diff
@@ -0,0 +1,415 @@
+  }
+
+  /**
+   * Add the Active Fields to the columnFieldSetters {@link ActiveFieldInfo}s
+   */
+  public void initColumnFieldSetters(List<FieldInfo> fieldInfos)
+  {
+    for (FieldInfo fi : fieldInfos) {
+      if (columnFieldSetters == null) {
+        columnFieldSetters = Lists.newArrayList();
+      }
+      columnFieldSetters.add(new AvroToPojo.ActiveFieldInfo(fi));
+    }
+  }
+
+  /**
+   * Sets the {@link FieldInfo}s. A {@link FieldInfo} maps a store column to a
+   * pojo field name.<br/>
+   * The value from fieldInfo.column is assigned to
+   * fieldInfo.pojoFieldExpression.
+   *
+   * @description $[].columnName name of the Output Field in POJO
+   * @description $[].pojoFieldExpression expression to get the respective field
+   *              from generic record
+   * @useSchema $[].pojoFieldExpression outputPort.fields[].name
+   */
+  public void setFieldInfos(List<FieldInfo> fieldInfos)
+  {
+    this.fieldInfos = fieldInfos;
+  }
+
+  /**
+   * Initialize the setters for generating the POJO
+   */
+  private void initializeActiveFieldSetters()
+  {
+    for (int i = 0; i < columnFieldSetters.size(); i++) {
+      ActiveFieldInfo activeFieldInfo = columnFieldSetters.get(i);
+
+      SupportType st = activeFieldInfo.fieldInfo.getType();
+
+      switch (st) {
+
+        case BOOLEAN:
+
+          activeFieldInfo.setterOrGetter = PojoUtils.createSetterBoolean(getCls(),
+              activeFieldInfo.fieldInfo.getPojoFieldExpression());
+          break;
+
+        case DOUBLE:
+          activeFieldInfo.setterOrGetter = PojoUtils.createSetterDouble(getCls(),
+              activeFieldInfo.fieldInfo.getPojoFieldExpression());
+          break;
+
+        case FLOAT:
+          activeFieldInfo.setterOrGetter = PojoUtils.createSetterFloat(getCls(),
+              activeFieldInfo.fieldInfo.getPojoFieldExpression());
+          break;
+
+        case INTEGER:
+          activeFieldInfo.setterOrGetter = PojoUtils.createSetterInt(getCls(),
+              activeFieldInfo.fieldInfo.getPojoFieldExpression());
+          break;
+
+        case STRING:
+          activeFieldInfo.setterOrGetter = PojoUtils.createSetter(getCls(),
+              activeFieldInfo.fieldInfo.getPojoFieldExpression(), activeFieldInfo.fieldInfo.getType().getJavaType());
+          break;
+
+        case LONG:
+          activeFieldInfo.setterOrGetter = PojoUtils.createSetterLong(getCls(),
+              activeFieldInfo.fieldInfo.getPojoFieldExpression());
+          break;
+
+        default:
+          activeFieldInfo.setterOrGetter = PojoUtils.createSetter(getCls(),
+              activeFieldInfo.fieldInfo.getPojoFieldExpression(), Byte.class);
+          break;
+      }
+
+      columnFieldSetters.get(i).setterOrGetter = activeFieldInfo.setterOrGetter;
+    }
+  }
+
+  /**
+   * Serializes a given object, Added as a plug to serialize non-primitive types
+   * 
+   * @return byte[]
+   */
+  private byte[] serialize(Object obj) throws IOException
```

Comment:
Can you please add class level javadoc comment, if this comes in action and how would the POJO get populated with it?